x-pack/plugin/apm-integration/src/main/java/org/elasticsearch/xpack/apm/APMTracer.java

```diff
@@ -81,7 +131,7 @@ public void onTaskUnregistered(Task task) {
 
     public static class CapturingSpanExporter implements SpanExporter {
 
-        private List<SpanData> capturedSpans = new ArrayList<>();
+        private List<SpanData> capturedSpans = Collections.synchronizedList(new ArrayList<>());
```

Comment:
This was populated in small chunks from multiple threads and loosing updates sometimes